cli/lib/kontena/cli/stacks/upgrade_command.rb

```diff
@@ -130,6 +195,11 @@ def execute
         abort_on_validation_errors(data[:local][:loader].reader.errors, data[:local][:loader].source)
       end
 
+      if dry_run?
+        dry_report(merged)
```

Comment:
Should this happen before the `unless removes.empty?` -> `merged.delete(removed_stack)`? ATM removing a child stack and doing `--dry-run` prompts for the stack removal, and then the `dry_report` never hits the `Stack #{stackname} would be removed from master` case:

```
kontena stack upgrade -v foo=asdf --dry-run deptest test/deptest.yaml 
 [done] Parsing test/deptest.yaml     
 [done] Reading stack deptest from master     

Stacks to be removed because they are no longer depended on:
- deptest-foo

Warning: This can not be undone, data will be lost.
> Destructive command. You can skip this prompt by running this command with --force option. Are you sure? Yes
DRY RUN: The following services would be kept:
- deptest/redis
- deptest-bar/redis
```

I think it would be more friendly if `--dry-run` didn't prompt about destructive commands that will lose data...

Reply:
Yep it was there for simulation purposes. Needs some reworking.